### ISSUE TITLE
Cannot pull images with rootless docker on openSUSE Tumbleweed

### ISSUE THREAD

**Issue by tilfischer** [rootless-containers/slirp4netns#329]
Dear all,

recently, I faced a strange issue with rootless docker on openSUSE Tumbleweed, which prevented any pulling of images. Basically, no traffic seems to be leaving the rootlesskit's namespace. Pulling an docker images always gives an i/o timeout:

```
someuser@somehost:~> docker pull hello-world
Using default tag: latest
Error response from daemon: Get "https://registry-1.docker.io/v2/": dial tcp: lookup registry-1.docker.io on 10.0.2.3:53: read udp 10.0.2.100:48971->10.0.2.3:53: i/o timeout
```

After quiet some digging and and with the help of two users at the [docker forum](https://forums.docker.com/t/rootless-docker-i-o-timeout-with-docker-pull/137848/24) and [openSUSE forum](https://forums.opensuse.org/t/rootless-docker-i-o-timeout-with-docker-pull/169468/4), the issue seems to be connected to slirp4netns: 

Following the [slirp4netns documentation](https://github.com/rootless-containers/slirp4netns/blob/master/slirp4netns.1.md):

```
Requires /etc/resolv.conf not to be a symlink to a file outside /etc and /run.
```

However, in openSUSE the resolve.conf is generated in /var/run/netconfig/ from what is given in /etc/sysconfig/network/config and the first is symlinked to /etc/resolve.conf. Removing the simlinked /etc/resolve.conf solved the issue, but should not survive any restart of the sever:

```
someuser@somehost:~> docker pull hello-world
Using default tag: latest
latest: Pulling from library/hello-world
719385e32844: Pull complete 
Digest: sha256:4f53e2564790c8e7856ec08e384732aa38dc43c52f02952483e3f003afbf23db
Status: Downloaded newer image for hello-world:latest
docker.io/library/hello-world:latest
```

Best

**Comment by AkihiroSuda** [comment]
Thanks for reporting.
Was it working before?
Was something changed on tumbleweed recently?


Adding some path (`/var/run` ?) to `--copy-up` helps?
https://github.com/moby/moby/blob/591c98cdbee73d36fe27abdb4456f8d70b8e549d/contrib/dockerd-rootless.sh#L107

**Comment by tilfischer** [comment]
Thank for your reply! 

I added `--copy-up=/var/run to the line mentioned and uninstalled and installed rootless docker. However, this did not help. 

What I have found out so far, is that netconfig was changed about a year ago, so that it no use /run/netconfig instead of /var/run/netconfig. 

Copying /var/run/netconfig/resolv.conf to /etc or adding a symlink there for /run/netconfig/resolv.conf to /etc/resolv.conf works fine. 

Hence, that seems not be an issue for slirp4netns.
